src/qutip_qip/circuit/text_renderer.py

```diff
@@ -0,0 +1,497 @@
+                self._render_strs["mid_frame"][wire] += mid_frame
+                self._render_strs["bot_frame"][wire] += mid_frame
+
+    def _update_qbridge(
+        self,
+        gate: Gate,
+        wire_list_control: List[int],
+        xskip: int,
+        width: int,
+        is_top_closer: bool,
+    ):
+        """
+        Update the render strings for part of the multi qubit gate drawn on the control wires.
+
+        Parameters
+        ----------
+        gate : Gate
+            The gate to be drawn.
+
+        wire_list_control : list
+            The list of control wires the gate is acting on and all the wires in between.
+
+        xskip : float
+            The horizontal value for getting to requested layer.
+
+        width : int
+            The width of the gate.
+
+        is_top_closer : bool
+            If the top of the gate is closer to the first control wires.
+        """
+
+        bar_conn = " " * (width // 2) + "│" + " " * (width // 2 - 1)
+        mid_bar_conn = "─" * (width // 2) + "│" + "─" * (width // 2 - 1)
+        node_conn = "─" * (width // 2) + "▇" + "─" * (width // 2 - 1)
+        self._adjust_layer(wire_list_control, xskip)
+
+        for wire in wire_list_control:
+            if wire not in gate.targets:
+                if wire in gate.controls:
+                    # check if the control wire is the first or last control wire.
+                    # used in cases of multiple control wires
+                    if (
+                        wire == wire_list_control[0]
+                        or wire == wire_list_control[-1]
+                    ):
+                        self._render_strs["mid_frame"][wire] += node_conn
+                        self._render_strs["top_frame"][wire] += (
+                            bar_conn if is_top_closer else " " * len(bar_conn)
+                        )
+                        self._render_strs["bot_frame"][wire] += (
+                            bar_conn
+                            if not is_top_closer
+                            else " " * len(bar_conn)
+                        )
+                    else:
+                        self._render_strs["top_frame"][wire] += bar_conn
+                        self._render_strs["mid_frame"][wire] += node_conn
+                        self._render_strs["bot_frame"][wire] += bar_conn
+                else:
+                    self._render_strs["top_frame"][wire] += bar_conn
+                    self._render_strs["mid_frame"][wire] += mid_bar_conn
+                    self._render_strs["bot_frame"][wire] += bar_conn
+
+    def _update_swap_gate(self, gate: Gate):
+        """
+        Update the render strings for the SWAP gate
+        """
+
+        wire_list = list(range(min(gate.targets), max(gate.targets) + 1))
+        layer = max(len(self._layer_list[i]) for i in wire_list)
+        xskip = self._get_xskip(wire_list, layer)
+        self._adjust_layer(wire_list, xskip)
```

Comment:
This part could be removed once ` _update_swap_gate` is moved to later part in `self.layout`